frontends/p4/uniqueNames.cpp

```diff
@@ -19,6 +19,31 @@ limitations under the License.
 
 namespace P4 {
 
+namespace {
+
+class FindActionCalls : public Inspector {
+    ReferenceMap* refMap;
+    TypeMap* typeMap;
+    RenameMap* renameMap;
+ public:
+    explicit FindActionCalls(ReferenceMap* refMap, TypeMap* typeMap, RenameMap* renameMap) :
+            refMap(refMap), typeMap(typeMap), renameMap(renameMap)
+    { CHECK_NULL(refMap); CHECK_NULL(typeMap); CHECK_NULL(renameMap); }
+
+    void postorder(const IR::MethodCallExpression* expression) {
+        auto mi = MethodInstance::resolve(expression, refMap, typeMap);
+        if (!mi->is<P4::ActionCall>())
+            return;
+        auto ac = mi->to<P4::ActionCall>();
+
+        auto table = findContext<IR::P4Table>();
+        if (table != nullptr)
+            renameMap->foundInTable(ac->action);
+    }
```

Comment:
This seems excessively verbose to me -- I think something like

    if (auto ac = MethodInstance::resolve(expression, refMap, typeMap)->to<P4::ActionCall>())
        if (findContext<IR::P4Table>())
            renameMap->foundInTable(ac->action);

is more readable.